### PR TITLE
fs: implement "New ..." button and the popup menu

### PR DESCRIPTION
**Pull request by songgao**

### CODE REVIEW

File: shared/fs/header/add-new.js [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    case 'public':
+      return 'icon-folder-public-16'
+    case 'team':
+      return 'icon-folder-team-16'
+    default:
+      return 'iconfont-question-mark'
+  }
+}
+
+const header = (pathElementsNoKeybase: Array<string>) => {
```
Comment (jzila) [inline shared/fs/header/add-new.js]: I'd prefer not to do this `pathElementsNoKeybase` thing. In all cases you have to slice it anyway (or take the first element), so you might as well just pass around the whole set of `pathElements`.

File: shared/fs/header/index.js [hunk 2]
```diff
@@ -0,0 +1,4 @@
+// @flow
+import FolderHeader from './container'
+
+export default FolderHeader
```
Comment (akalin-keybase) [inline shared/fs/header/index.js]: I think the usual style is to put the unconnected component in `foo/index.js` and the connected component in `foo/container.js`, and have everything external use the connected one

File: shared/fs/header/add-new.js [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+              <Icon
+                type={getIcon(pathElementsNoKeybase[0])}
+                color={globalColors.blue}
+                style={stylesIconFolderType}
+                key="icon"
+              />,
+              <Text key="text" type="BodySmallSemibold">
+                {pathElementsNoKeybase[0]}
+              </Text>,
+              ...[].concat(
```
Comment (akalin-keybase) [inline shared/fs/header/add-new.js]: i think you can just have `...pathElementsNokeybase.slice(1).map(...)`, right?

File: shared/fs/header/add-new.js [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+      return 'icon-folder-public-16'
+    case 'team':
+      return 'icon-folder-team-16'
+    default:
+      return 'iconfont-question-mark'
+  }
+}
+
+const header = (pathElementsNoKeybase: Array<string>) => {
+  return !isMobile
```
Comment (akalin-keybase) [inline shared/fs/header/add-new.js]: `isMobile && ...`?

File: shared/fs/header/add-new.js [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+                ])
+              ),
+            ]}
+          </Box>
+        ),
+      }
+}
+
+const AddNew = (props: AddNewProps & FloatingMenuParentProps) => {
+  return !props.menuItems.length ? null : (
```
Comment (akalin-keybase) [inline shared/fs/header/add-new.js]: `props.menuItems.length && ...`?

File: shared/fs/header/container.js.flow [hunk 6]
```diff
@@ -0,0 +1,6 @@
+// @flow
+import * as React from 'react'
+
+type FolderHeaderProps = any
```
Comment (akalin-keybase) [inline shared/fs/header/container.js.flow]: I think you can type this as `{path: Array<string>}`

File: shared/fs/header/index.js (hunk 2 above)
Comment (akalin-keybase) [reply shared/fs/header/index.js]: Ah, I see, both the unconnected and the connected versions are split platform. I suggest `foo/index.{desktop,native}.js` for the unconnected and `foo/container.{desktop,native}.js` + `foo/container.js.flow` for the connected, then

**Comment by akalin-keybase** [comment]
Forgot to mention, stories would be great, but that can be in a future PR

File: shared/fs/header/index.js (hunk 2 above)
Comment (songgao) [reply shared/fs/header/index.js]: Ah I see. Will do, but would you mind if I address these in a separate PR? Was worried it's gonna cause a lot of conflict when rebasing.

File: shared/fs/header/container.native.js [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+
+const mapDispatchToProps = (dispatch: Dispatch) => ({
+  onBack: () => dispatch(navigateUp()), // TODO: put if on route ...
+})
+
+const mergeProps = (stateProps, {onBack}, {path}) => {
+  const elems = Types.getPathElements(path)
+  return {
+    path,
+    title: elems.length > 1 ? elems.pop() : 'Keybase Files',
```
Comment (akalin-keybase) [inline shared/fs/header/container.native.js]: don't think you want to do `pop`, as that modifies `elems`. I think you want `elems.length > 1 ? elems[elems.length - 1] : ...`.

Also, did you mean `elems.length > 0`?

Comment (songgao) [reply shared/fs/header/container.native.js]: I meant to do `> 1`, since when it only has one element, it's `/keybase` and we'd be showing "Keybase Files". Will swap `pop()` out as that's probably cheaper. Thanks!

File: shared/fs/header/index.js [hunk 8]
```diff
@@ -0,0 +1,7 @@
+// @flow
+import FolderHeader from './container'
```
Comment (jzila) [inline shared/fs/header/index.js]: Why is this file necessary? I.e. why can't callers just import from `header/container`?

Comment (songgao) [reply shared/fs/header/index.js]: Good point. I'm gonna reorganize these a bit after these PRs are merged. Fred also suggested some naming schemes above.

Comment (jzila) [reply shared/fs/header/index.js]: We've been doing a lot of renaming lately. It creates conflicts when we're working on the same files. Let's try to minimize how often this has to happen.

Comment (songgao) [reply shared/fs/header/index.js]: Some of the renamings here are because some types are getting bloated and I wanted to split them up.
